docs/wallet_browser_extension/provider/api_documentation.md

```diff
@@ -0,0 +1,448 @@
+---
+tags:
+  - wallet
+  - browser extension
+  - provider API
+---
+
+# API Documentation
+
+## Properties
+
+### isXPLL
+
+This property is true if the user has extension installed.
+
+```js
+window.xpll.isXPLL; // => true
+```
+
+## Request
+
+The request method is intended as a transport- and protocol-agnostic wrapper function for [Remote Procedure Calls (RPCs)][1]
+
+If resolved, the Promise will resolve with a result per the RPC method’s specification.
+If the returned Promise rejects,
+it will reject with a `XPLLProviderRpcError` as specified in the [Errors][2] section below.
+
+```ts
+interface RequestArguments {
+  readonly method: string;
+  readonly params?: readonly unknown[] | object;
+}
+```
+
+```ts
+request(data: RequestArguments): Promise<Result>;
+// or reject with `XPLLProviderRpcError`
+```
+
+### Parameters
+
+- `method`: Method indicates which RPC method to call.
+- `params`: Parameters is an optional array or object of arguments to pass with the RPC.
+
+### Helper Functions
+
+To improve the developer experience,
+the provider also implemented a list of helper functions,
+which helps the user simplify the RPC method call.
+
+## Get Accounts
+
+Get the list of [addresses][3] for the accounts owned by the user.
+
+!!! warning
+    The `get_accounts` should have permission granted; otherwise, the `User Rejected Request 4001` error will be thrown.
+    Please check [Permission][4] for more detail.
+
+
+```ts
+interface GetAccountsRequest {
+  method: "get_accounts";
+}
+
+type GetAccountsResult = string[];
+```
+
+### Example of usage
```

Comment:
I think "Example of usage" doesn't need to be a section because this looks redundant in the table of contents. Same applies for others examples.